src/lib.rs

```diff
@@ -432,4 +601,68 @@ mod test {
         assert!(txt.is_global_group());
         assert!(!txt.member_of("telescopes"));
     }
+
+    #[test]
+    fn signing_and_verifying() {
+        let invoice = r#"
+        bindleVersion = "1.0.0"
+
+        [bindle]
+        name = "aricebo"
+        version = "1.2.3"
+
+        [[parcel]]
+        [parcel.label]
+        sha256 = "aaabbbcccdddeeefff"
+        name = "telescope.gif"
+        mediaType = "image/gif"
+        size = 123_456
+        
+        [[parcel]]
+        [parcel.label]
+        sha256 = "111aaabbbcccdddeee"
+        name = "telescope.txt"
+        mediaType = "text/plain"
+        size = 123_456
+        "#;
+
+        let mut invoice: crate::Invoice = toml::from_str(invoice).expect("a nice clean parse");
+
+        // Base case: No signature, no keyring should pass.
+        assert!(invoice.signature.is_none());
+        invoice
+            .verify(vec![])
+            .expect("If no signature, then this should verify fine");
+
+        // Create two signing keys.
+        let mut rng = rand::rngs::OsRng {};
+        let keypair1 = Keypair::generate(&mut rng);
+        let signer_name1 = "Matt Butcher <matt@example.com>".to_owned();
+
+        let keypair2 = Keypair::generate(&mut rng);
+        let signer_name2 = "Not Matt Butcher <not.matt@example.com>".to_owned();
+
+        // Put one of the two keys on the keyring
+        let keyring = vec![keypair2.public.clone()];
+
+        // Add two signatures
+        invoice
+            .sign(signer_name1, SignatureRole::Creator, keypair1)
+            .expect("sign the parcela");
```

Comment:
s/parcela/parcel?

Reply:
(As opposed to the Spanish and Romanian word for "piece of land"?)

Reply:
Yes, we'll switch it to the english version of "a piece of land"

Reply:
fixed in both places.